assets/src/components/ui/toggle-button/toggle-button.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * ToggleButton
+ * Returns a component that can be used to trigger an action like opening or closing a container.
+ * The component appears as an icon only but acts like a button.
+ *
+ * @function
+ * @param {string} htmlId               Used to generate an HTML "id" attribute for all components.
+ *                                      Additional text is appended by each component.
+ * @param {string} htmlClass            Used to generate an HTML "class" attribute for all components.
+ *                                      Additional text is appended by each component.
+ * @param {string} description          Two or three words (hyphenated) that describe the container
+ *                                      that the ToggleButton is associated with.
+ * @param {Function} toggleCallback 	Function called when the "button" component is triggered
+ * @param {string} dashicon             Css class that defines the dashicon icon
+ * @param {Number} tabIndex             tab index for the element that controls when it receives focus
+ * @return {Object}                     "button" for opening the modal container
+ */
+export const ToggleButton = (
+	{
+		htmlId,
+		htmlClass,
+		description,
+		toggleCallback,
+		dashicon = 'no',
+		tabIndex = 0,
+	}
+) => {
+	return toggleCallback && (
+		<div
+			id={ `${ htmlId }-${ description }` }
+			className={ `${ htmlClass }-${ description } ee-${ description } ee-toggle-button` }
+			onClick={ ( event ) => toggleCallback( event ) }
+			onKeyPress={ ( event ) => toggleCallback( event ) }
```

Comment:
I think you should use `onKeyDown` here instead.  `onKeyPress` ignores certain keys (shift/ctrl etc).  So if someone wants to bind those keys to this button they won't be able to.

Also, I'm wary about you calling the same callback for the `onClick` and `onKeyPress` events here.  That could lead to some unexpected behaviour for the implementor.  Instead you should look for specific `onClick` and `onKeyPress` (or if you change it, `onKeyDown`) props.  Don't create a new named prop for something that is already familiar.  It also makes it very clear what events will get triggered by this component.

Reply:
> onKeyPress ignores certain keys (shift/ctrl etc). 

yes because that allows you to listen for key combinations like `ALT`+`x` instead of firing separately after each `onKeyDown` event.
So conversely, using `onKeyDown` would prevent someone from binding key combinations from being used, so changing to `onKeyDown` has its detractors as well right? Question is, how often do you press (shift/ctrl etc) on their own and expect anything to happen?

But anyways, the purpose here is to simply trigger the firing of the callback any time the button is clicked, or any time a key is pressed when the button has keyboard focus, which is satisfied by using a combination of `onClick` and `onKeyPress`. 

Plz keep in mind that not everything we build has to be approached from the perspective of being some completely abstracted library that is intended for release to the general public and therefore needs to satisfy every possible use case and scenario. We are not WordPress. It is perfectly acceptable for us to develop some tools that are only intended for our own purposes at the moment, and IF use by third parties becomes apparent, then those tools can be improved if need be. For now, I am only adding these components so that I can achieve other goals (REM). Our primary objective is not to build a library of generic tools for release on npm.

Reply:
> Plz keep in mind that not everything we build has to be approached from the perspective of being some completely abstracted library that is intended for release to the general public and therefore needs to satisfy every possible use case and scenario.

Then don't add this to our `@eventespresso/components` library which is intended for re-use.  Just do what you need for REM in REM's own component library and once you've determined the appropriate form for re-usability then we can add something to @eventespresso/components.  

I _am_ going to be viewing/evaluating things added to `@eventespresso/components` from the standpoint of reusability not only within EE core but also across other add-ons.

So, if your answers for my line of queries are, "I'm only doing this because REM needs it this way" and thus it has a very opinionated behaviour, just keep the component in the add-on you are using it in.  However if you're willing to consider how the component might get used in EE core, or other add-ons, then let's iterate further on this in here.

I'm completely fine with some components living in their own add-ons for a while if necessary.